x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ml/dataframe/DataFrameAnalyticsConfigTests.java

```diff
@@ -88,6 +90,49 @@ protected DataFrameAnalyticsConfig createTestInstance() {
         return createRandom(randomValidId(), lenient);
     }
 
+    @Override
+    protected List<Version> bwcVersions() {
+        return Arrays.asList(Version.V_7_0_0,
+            Version.V_7_1_1,
+            Version.V_7_2_1,
+            Version.V_7_3_2,
+            Version.V_7_4_2,
+            Version.V_7_5_3,
+            Version.V_7_6_0,
+            Version.V_7_7_0);
+    }
+
+    @Override
+    protected DataFrameAnalyticsConfig mutateInstanceForVersion(DataFrameAnalyticsConfig instance, Version version) {
+        DataFrameAnalyticsConfig.Builder builder = new DataFrameAnalyticsConfig.Builder(instance);
+        if (version.before(Version.V_7_6_0)) {
+            if (instance.getAnalysis() instanceof Regression) {
+                // TODO regression and classification serialize a randomized value for the random seed
```

Comment:
Do these belong in the respective test classes? See below.

Reply:
This `todo` shows that it is impossible to accurately test equality for `Regression` and `Classification`. 

When reading in from an older version, a random seed is generated. There is no way of determining which value that is so that you can accurately say "this is what the object would be if serialized from an old version".

Reply:
Added some tests to classification and regression analysis types. 

Also updated the scaffolding to better handle unique situations like the random seed generation.